cogito/protocol.go

```diff
@@ -49,6 +94,27 @@ type PutRequest struct {
 	Env    Environment
 }
 
+// NewPutRequest returns a [PutRequest] ready to be used.
+func NewPutRequest(input []byte) (PutRequest, error) {
```

Comment:
So we are forced to use multiple functions because the struct differs.

Reply:
Yes, each time the JSON input is partially different. "Source" is always there but not the rest.
See https://concourse-ci.org/implementing-resource-types.html

Reply:
For my understanding why the difference:
`func NewGetRequest(input []byte) (*GetRequest, error) {`
vs
`func NewPutRequest(input []byte) (PutRequest, error) {`
the first one returns the pointer, but the second full structure?

Reply:
Thanks for the comments and links. I am going through them now.

> So in this case, `func NewGetRequest() *GetRequest` returns a pointer to the memory allocated (on the heap) (in C this would crash instead) from the function, and `func NewPutRequest() PutRequest` returns a copy of the memory allocated (on the stack) from the function.

But I still don't get why the difference in this code? Why this "inconsistency"?

Reply:
> But I still don't get why the difference in this code?

I misunderstood the question. Now I see. There is no reason indeed, so I am adding a commit to remove the difference, thanks for asking.